File: backend/app/schemas/diary.py
"""AI 골프 일기 응답 스키마.

생성 요청은 JSON이 아니라 multipart/form-data(텍스트 또는 오디오 파일)라 별도의
DiaryCreate 모델 없이 라우터에서 Form/File을 직접 받는다.
"""
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DiaryListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    round_id: int | None
    summary: str
    mood: str
    created_at: datetime


class DiaryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    round_id: int | None
    round_summary: str | None = None
    raw_text: str
    summary: str
    mood: str
    highlights: str
    improvement_points: str
    next_goal: str
    created_at: datetime

File: frontend/src/types/diary.ts
export interface DiaryListItem {
  id: number;
  round_id: number | null;
  summary: string;
  mood: string;
  created_at: string;
}

export interface Diary extends DiaryListItem {
  round_summary: string | null;
  raw_text: string;
  highlights: string;
  improvement_points: string;
  next_goal: string;
}

export interface DiaryCreatePayload {
  text?: string;
  audioBlob?: Blob;
  roundId?: number;
}
